src/include/optimizer/stats/cost.h

```diff
@@ -137,6 +145,80 @@ class Cost {
   inline static double InnerHashJoin() { return DEFAULT_COST; }
 
   /*
+   * Join with Sampling
+   */
+  static double InnerNLJoinWithSampling(
```

Comment:
Why not add those functions in InnerNLJoin / InnerHashJoin and use a flag to indicate if we want to use sampling? Then the cost computation can be shared and it will be much easier to switch the configuration for the system.

Reply:
I think it's a great idea. I'll do that.